### DISCUSSION TITLE
CharacterSet Auto Selector

### DISCUSSION THREAD

**Discussion by DahyeonWoo** [euphony-io/euphony#221] [Ideas]
Hello, I am planning to make CharacterSet Auto Selector.

## File
euphony/euphony/src/main/cpp/core/charset
* CharsetAutoSelector.h
* CharsetAutoSelector.cpp
* README.md

euphony/euphony/src/main/cpp/tests/charset
* CharsetAutoSelectorTest.cpp

Is there another name? 😀 

## Input & Output
* Input: desired String value
* Output: Return the charset showing the shortest encoding result (Charset interface)

## Unit Test
* gtest

## Docs
* Charset, ASCII, UTF, UNICODE
  * DefaultCharset, ASCIICharset, UTF-8, UTF-16, UTF-32

**Comment by designe** [comment]
I think name is very intuitive :)
It'll be super helpful to euphony users!

Let me create issue from this ;) Thanks for suggestion!
